app/src/main/java/com/github/orkest/domain/FireStoreDatabaseAPI.kt

```diff
@@ -311,6 +299,47 @@ class FireStoreDatabaseAPI {
         return future
     }
 
+
+    //TODO: Add the condition on posts belonging to the user's friends
+    /**
+     * Returns the list of posts that were published after the last connection of the user
+     * and from the last [maxDaysNb] days
+     * @param lastConnection the last time the user was connected
+     * @param maxDaysNb the max number of days we want to get the posts from, default and max value is 30
+     * @return a completable future of a list of post that were recently published
+     */
+    fun getRecentPostsFromDataBase(lastConnection: LocalDateTime, maxDaysNb: Long = DEFAULT_MAX_RECENT_DAYS): CompletableFuture<List<Post>> {
+        //Preconditions to check
+        require(maxDaysNb <= DEFAULT_MAX_RECENT_DAYS) { "maxDaysNb must be less or equal to $DEFAULT_MAX_RECENT_DAYS" }
+
+        val futures = mutableListOf<CompletableFuture<List<Post>>>()
+        val maxDaysAgo = LocalDateTime.now().minusDays(maxDaysNb)
+        val recentPostsTime = if (lastConnection.isAfter(maxDaysAgo)) lastConnection else maxDaysAgo
+
+        val listMonths  = mutableListOf<Int>(recentPostsTime.monthValue)
+        //If the recentPostsTime is not in the current month, we need to add the next month
+        if (recentPostsTime.monthValue != LocalDateTime.now().monthValue)
+            if (recentPostsTime.monthValue == 12)
+                listMonths.add(1)
+            else
+                listMonths.add(recentPostsTime.monthValue+1)
+
+        //Get all posts that were published after the recentPostsTime - One query per month
+        listMonths.forEach { month ->
+            run {
+                futures.add(recentPostsQuery(month,
+                    recentPostsTime.year,
+                    if (month == recentPostsTime.monthValue) recentPostsTime.dayOfMonth else 1))
+            }
+        }
```

Comment:
Why did you need to wrap your code in a run block  ? Did you have an error when not doing so ?

Reply:
Yes, without it, I get a warning mentioning that the lambda code was just ignored